uniprotkb-rest/src/main/java/org/uniprot/api/uniprotkb/service/UniProtEntryService.java

```diff
@@ -86,54 +100,34 @@ public UniProtEntry getByAccession(String accession, String fields) {
         }
     }
 
-    public Stream<UniProtEntry> stream(SearchRequestDTO request) {
-        SolrRequest solrRequest = createSolrRequest(request, false);
-        return storeStreamer.idsToStoreStream(solrRequest);
-    }
-
-    public Stream<String> streamIds(SearchRequestDTO request) {
-        SolrRequest solrRequest = createSolrRequest(request, false);
-        return storeStreamer.idsStream(solrRequest);
-    }
-
-    public Stream<String> streamRDF(SearchRequestDTO searchRequest) {
-        SolrRequest solrRequest = createSolrRequest(searchRequest, false);
+    public Stream<String> streamRDF(SearchRequest searchRequest) {
+        SolrRequest solrRequest = createDownloadSolrRequest(searchRequest);
         return this.storeStreamer.idsToRDFStoreStream(solrRequest);
     }
 
-    private SolrRequest createSolrRequest(SearchRequestDTO request, boolean includeFacets) {
-        SolrRequest.SolrRequestBuilder requestBuilder = SolrRequest.builder();
-        requestBuilder.rows(request.getSize());
-        requestBuilder.queryBoosts(queryBoosts);
+    @Override
+    protected SolrRequest createSolrRequest(SearchRequest request, boolean includeFacets) {
 
-        String requestedQuery = request.getQuery();
+        UniProtKBRequest uniProtRequest = (UniProtKBRequest) request;
+        // fill the common params from the basic service class
+        SolrRequest solrRequest = super.createSolrRequest(uniProtRequest, includeFacets);
 
-        if (needsToFilterIsoform(request)) {
-            requestBuilder.filterQuery(UniProtField.Search.is_isoform.name() + ":" + false);
-        }
+        // uniprotkb related stuff
+        solrRequest.setQueryBoosts(queryBoosts);
 
-        if (request.isShowMatchedFields()) {
-            requestBuilder.termQuery(requestedQuery);
-            uniProtTermsConfig.getFields().forEach(requestBuilder::termField);
+        if (needsToFilterIsoform(uniProtRequest)) {
+            solrRequest.setFilterQueries(
```

Comment:
Is there any specific reason that you replaced requestBuilder.filterQuery to solrRequest.setFilterQueries? I think that is safer at this point that we are adding things to super.createSolrRequest to just add one filter query, than replace (set) it. Does it make sense to you?

Reply:
Good catch. I fixed it.
`   if (needsToFilterIsoform(uniProtRequest)) {
-            solrRequest.setFilterQueries(
-                    Arrays.asList(UniProtField.Search.is_isoform.name() + ":" + false));
+            solrRequest.getFilterQueries().add(UniProtField.Search.is_isoform.name() + ":" + false);
         }`